components/service/glean/scripts/sdk_generator.gradle

```diff
@@ -142,6 +187,18 @@ ext.gleanGenerateMetricsAPI = {
     def generateKotlinAPI = task("generateMetricsSourceFor${variant.name.capitalize()}", type: Exec) {
         description = "Generate the Kotlin code for the Metrics API"
 
+        if (project.ext.has("allowMetricsFromAAR")) {
+            // This is sufficiently lazy to be valid at configuration time.  See the model at
+            // https://github.com/google/protobuf-gradle-plugin/blob/6d99a421c8d15710045e4e8d31a3af6cb0cc3b70/src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy#L270-L277
+            inputs.files variant.compileConfiguration.incoming.artifactView {
```

Comment:
Just FYI, if you did want to restrict to only `geckoview-*` dependencies, I think you'd probably do it here, filtering by dependency name or group or something.

Reply:
Thanks for mentioning that. I thought about doing this, but I really don't want to hardcode too many GeckoView specific things in the Glean SDK. If we stumble upon the situation of multiple AARs exporting metrics.yaml files, this won't break compilation, but will generate code that will end up being stripped away by r8/PG. Worst case, we will end up introducing another configuration option to restrict this. I'd leave this as it is for now, for the current use-case.